pkg/oci_helper/oci_helper.go

```diff
@@ -0,0 +1,326 @@
+	}
+	tagged, ok := repo.(reference.Tagged)
+	if !ok {
+		return "latest", nil
+	}
+	return tagged.Tag(), nil
+}
+
+func GetRepositoryFromImage(image string) (string, error) {
+	repo, err := reference.Parse(image)
+	if err != nil {
+		return "", fmt.Errorf("parse image %q: %w", image, err)
+	}
+	if named, ok := repo.(reference.Named); ok {
+		return named.Name(), nil
+	}
+	return "", fmt.Errorf("image has to be a named reference")
+}
+
+func NormalizeImage(image string) (string, error) {
+	name, err := reference.ParseNormalizedNamed(image)
+	if err != nil {
+		return "", fmt.Errorf("parse normalized image %q: %w", image, err)
+	}
+	return reference.TagNameOnly(name).String(), nil
+}
+
+func GetHostString(repository string) (string, error) {
+	repo, err := reference.Parse(repository)
+	if err != nil {
+		return "", fmt.Errorf("parse repository %q: %w", repository, err)
+	}
+	if named, ok := repo.(reference.Named); ok {
+		return reference.Domain(named), nil
+	}
+	return "", fmt.Errorf("image has to be a named reference")
+}
+
+func SetupAuthVariablesAndFlags(cmd *cobra.Command, authOptions *AuthOptions) {
+	// Flag inspired by https://github.com/containers/common/blob/cac40138f7e3c2b29ca32e64348535516bf6aa51/pkg/auth/cli.go#L48
+	cmd.Flags().StringVar(&authOptions.AuthFile, "authfile", DefaultAuthFile,
+		"path of the authentication file. This overrides the REGISTRY_AUTH_FILE environment variable")
+	viper.BindPFlag("registry.auth_file", cmd.Flags().Lookup("authfile"))
+	viper.BindEnv("registry.auth_file", "REGISTRY_AUTH_FILE")
+
+	cmd.Flags().BoolVar(&authOptions.Insecure, "insecure", false, "allow connections to HTTP only registries")
+}
+
+func CreateAuthClient(repository string, authOptions *AuthOptions) (*oras_auth.Client, error) {
+	logrus.Debugf("Using auth file %q", authOptions.AuthFile)
+
+	var cfg *configfile.ConfigFile
+	var err error
+
+	// 1. Explicitly setting the auth file
+	// 2. Using the default auth file
+	// 3. Using the default docker auth file if 2. doesn't exist
+	if authOptions.AuthFile != DefaultAuthFile {
+		authFileReader, err := os.Open(authOptions.AuthFile)
+		if err != nil {
+			return nil, fmt.Errorf("open auth file %q: %w", authOptions.AuthFile, err)
+		}
+		defer authFileReader.Close()
+		cfg, err = config.LoadFromReader(authFileReader)
+		if err != nil {
+			return nil, fmt.Errorf("load auth config: %w", err)
+		}
+	} else if _, err := os.Stat(authOptions.AuthFile); err == nil {
+		authFileReader, err := os.Open(authOptions.AuthFile)
+		if err != nil {
+			return nil, fmt.Errorf("open auth file %q: %w", authOptions.AuthFile, err)
+		}
+		defer authFileReader.Close()
+		cfg, err = config.LoadFromReader(authFileReader)
+		if err != nil {
+			return nil, fmt.Errorf("load auth config: %w", err)
+		}
+	} else {
+		logrus.Debugf("Couldn't find default auth file %q...", authOptions.AuthFile)
+		logrus.Debugf("Using default docker auth file instead")
+		logrus.Debugf("$HOME: %q", os.Getenv("HOME"))
```

Comment:
Could you please try to describe what's your idea there? I can't see how it'll work, if `authOptions.AuthFile == DefaultAuthFile` and DefaultAuthFile doesn't exist it'll fail on `os.Open(authOptions.AuthFile)`, but it should fallback to `cfg, err = config.Load("")`, right?

Reply:
Perhaps something like is more concise:

```go
	// 1. Explicitly setting the auth file
	// 2. Using the default auth file
	// 3. Using the default docker auth file if 2. doesn't exist
	authFileReader, err := os.Open(authOptions.AuthFile)
	if err != nil {
		if errors.Is(err, os.ErrNotExist) && authOptions.AuthFile == DefaultAuthFile {
			logrus.Debugf("Couldn't find default auth file %q...", authOptions.AuthFile)
			logrus.Debugf("Using default docker auth file instead")
			logrus.Debugf("$HOME: %q", os.Getenv("HOME"))

			cfg, err = config.Load("")
			if err != nil {
				return nil, fmt.Errorf("load auth config: %w", err)
			}
		}

		return nil, fmt.Errorf("open auth file %q: %w", authOptions.AuthFile, err)
	} else {
		defer authFileReader.Close()
		cfg, err = config.LoadFromReader(authFileReader)
		if err != nil {
			return nil, fmt.Errorf("load auth config: %w", err)
		}
	}
```

@burak-ok would you mind checking if that's equivalent to the previous one?

Reply:
I think that my thought was the following while implementing this:
1. If the user sets the auth file explicitly, i don't want to implicitly use the docker auth file
2. If the auth file is not set and not there, we use the docker auth implicitly.

Now I am thinking that we could also add an explicit flag that we want to use docker auth directly

WDYT?

Regarding the code suggestion from @mauriciovasquezbernal:
I think that is okay. I would remove the top comment `//1//2//3....` and add a single comment:
```golang
if err != nil {
 		// If the AuthFile was not set explicitly, we allow to fall back to the docker auth
		// since explicitly setting it means that the user only wants that specific auth file
		if !errors.Is(err, os.ErrNotExist) || authOptions.AuthFile != DefaultAuthFile {
			return nil, fmt.Errorf("open auth file %q: %w", authOptions.AuthFile, err)
		}
```